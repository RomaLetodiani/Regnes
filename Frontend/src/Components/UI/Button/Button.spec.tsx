import { render, screen } from "@testing-library/react";
import Button from "./Button";

describe("Button Component", () => {
  test("renders with primary styles by default", () => {
    render(<Button>Click Me</Button>);
    const button = screen.getByRole("button", { name: /click me/i });

    expect(button).toBeInTheDocument();
    expect(button).toHaveClass("bg-primary text-white");
  });

  test("renders with secondary styles when btnType is secondary", () => {
    render(<Button btnType="secondary">Click Me</Button>);
    const button = screen.getByRole("button", { name: /click me/i });

    expect(button).toBeInTheDocument();
    expect(button).toHaveClass("bg-secondary-200 text-primary");
  });

  test("renders with danger styles when btnType is danger", () => {
    render(<Button btnType="danger">Click Me</Button>);
    const button = screen.getByRole("button", { name: /click me/i });

    expect(button).toBeInTheDocument();
    expect(button).toHaveClass("bg-danger text-white");
  });

  test("applies additional class names", () => {
    render(<Button className="additional-class">Click Me</Button>);
    const button = screen.getByRole("button", { name: /click me/i });

    expect(button).toBeInTheDocument();
    expect(button).toHaveClass("additional-class");
  });

  test("renders with disabled attribute", () => {
    render(<Button disabled>Click Me</Button>);
    const button = screen.getByRole("button", { name: /click me/i });

    expect(button).toBeDisabled();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Button>Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
