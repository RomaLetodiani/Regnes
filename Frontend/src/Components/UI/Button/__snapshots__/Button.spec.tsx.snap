// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="rounded-lg cursor-pointer px-5 py-2 shadow-lg hover:opacity-70 bg-primary text-white"
  >
    Button
  </button>
</DocumentFragment>
`;
